### PR TITLE
Refactor distributions

### PR DESCRIPTION
**Pull request by alicanb**
Change parameter names so they are consistent with tensorflow. New distribution signatures:
- `Beta(concentration1, concentration0)`
- `Normal(loc, scale)`
- `Dirichlet(concentration)`
- `Gamma(concentration, rate)`

### CODE REVIEW

**Comment by Kaixhin** [comment]
Can we keep the previous names documented as alternative names for these parameters? This would make the docs more comprehensive.

File: torch/distributions/normal.py [hunk 1]
```diff
@@ -11,46 +10,46 @@
...
     r"""
     Creates a normal (also called Gaussian) distribution parameterized by
-    `mean` and `std`.
+    `loc` and `scale`.
 
     Example::
 
         >>> m = Normal(torch.Tensor([0.0]), torch.Tensor([1.0]))
-        >>> m.sample()  # normally distributed with mean=0 and stddev=1
+        >>> m.sample()  # normally distributed with loc=0 and scaledev=1
```
Comment (apaszke) [inline torch/distributions/normal.py]: nit: scaledev -> scale

**Review by apaszke — APPROVED** [review]
I agree it would be nice to add "(often referred to as alpha)" in the docs

File: torch/distributions/gamma.py [hunk 2]
```diff
@@ -16,18 +16,20 @@ def _standard_gamma(concentration):
...
 
         >>> m = Gamma(torch.Tensor([1.0]), torch.Tensor([1.0]))
-        >>> m.sample()  # Gamma distributed with shape concentration=1 and rate rate=1
+        >>> m.sample()  # Gamma distributed with concentration=1 and rate=1
          0.1046
         [torch.FloatTensor of size 1]
 
     Args:
         concentration (float or Tensor or Variable): shape parameter of the distribution
+        (often referred to as alpha)
```
Comment (apaszke) [inline torch/distributions/gamma.py]: I think you need to indent those lines, or they will render incorrectly

Comment (alicanb) [reply torch/distributions/gamma.py]: like this?
```
concentration1 (float or Tensor or Variable): 1st concentration parameter of the distribution
                                              (often referred to as alpha)

Comment (apaszke) [reply torch/distributions/gamma.py]: Just 4 spaces is enough

**Review by apaszke — APPROVED** [review]

**Pull request merged**